uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -762,25 +822,39 @@ a hash with all the checks that need to be applied on that specific scan type
 and severity.
 
 INPUTS:
-  - $scan_type: scan type of the file
-  - $severity : severity of the checks we want to loop through (exclude or warning)
-  - $headers  : list of different headers found in the C<mri_protocol_checks>
-                table for a given scan type
-  - $file     : file information hash ref
+  - $scan_type   : scan type of the file
+  - $severity    : severity of the checks we want to loop through (exclude or warning)
+  - $headers     : list of different headers found in the C<mri_protocol_checks>
+                   table for a given scan type
+  - $file        : file information hash ref
+  - $projectID   : candidate's project ID
+  - $subprojectID: session's subproject ID
+  - $visitLabel  : session name
 
 RETURNS: a hash with all information about the checks for a given scan type
 and severity
 
 =cut
 
 sub loop_through_protocol_violations_checks {
-    my ($this, $scan_type, $severity, $headers, $file) = @_;
+    my ($this, $scan_type, $severity, $headers, $file, $projectID, $subprojectID, $visitLabel) = @_;
 
     my %valid_fields; # will store all information about what fails
 
-    # query to fetch list of valid protocols in mri_protocol_checks table
-    my $query = "SELECT * FROM mri_protocol_checks "
-                . "WHERE Scan_type=? AND Severity=? AND Header=?";
+    # query to fetch list of valid protocol checks in mri_protocol_checks table
+    my $query = "SELECT * FROM mri_protocol_checks mpc "
+        . "JOIN mri_protocol_checks_group_rel mpcgr ON (mpc.ID=mpcgr.MriProtocolChecksGroupID) "
+        . "JOIN mri_protocol_checks_group_target mpcgt USING(MriProtocolChecksGroupID) "
+        . "WHERE mpc.Scan_type=? AND mpc.Severity=? AND mpc.Header=?";
+    $query .= defined $projectID
```

Comment:
same question as for the mri_protocol table. What happens if a project has a stricter t1 protocol than the generic t1 protocol?